### PR TITLE
chore: Resolve RemovedInDjango40Warning warnings.

### PR DESCRIPTION
**Pull request by christopappas**
**Description:**

Use `django.urls` instead of `django.conf.urls` to resolve REmovedInDjango40Warnings

**JIRA:**

No associated ticket.

**Additional Details**

* **Dependencies:**: List dependencies on other outstanding PRs, issues, etc.
* **Merge deadline:** List merge deadline (if any)
* **Testing instructions:** Provide non-trivial testing instructions
* **Author concerns:** List any concerns about this PR

**Reviewers:**
- [x] @edx/arch-review
- [x] tag reviewer

**Merge checklist:**
- [ ] All reviewers approved
- [x] CI build is green
- [x] Version bump if needed
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [x] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

**Review by macdiesel — APPROVED** [review]

**Review by dianakhuang — APPROVED** [review]

File: auth_backends/urls.py [hunk 1]
```diff
@@ -2,15 +2,16 @@
 
 Add these to your project's `urlpatterns` to avoid duplicating code.
 """
-from django.conf.urls import url, include
+from django.conf.urls import include
+from django.urls import re_path
```
Comment (jmbowman) [inline auth_backends/urls.py:6]: `include` is also now available via `django.urls`, so this whole thing can be simplified to `from django.urls import include, re_path`.

File: auth_backends/urls.py [hunk 2]
```diff
@@ -2,15 +2,16 @@
...
     EdxOAuth2LogoutView,
 )
 
 oauth2_urlpatterns = [
-    url(r'^login/$', EdxOAuth2LoginView.as_view(), name='login'),
-    url(r'^logout/$', EdxOAuth2LogoutView.as_view(), name='logout'),
-    url('', include('social_django.urls', namespace='social')),
+    re_path(r'^login/$', EdxOAuth2LoginView.as_view(), name='login'),
+    re_path(r'^logout/$', EdxOAuth2LogoutView.as_view(), name='logout'),
+    re_path('', include('social_django.urls', namespace='social')),
```
Comment (jmbowman) [inline auth_backends/urls.py:16]: We'll probably want to change these to use `path()` later, but I'm fine with punting that until we have more experience and docs for doing it.

**Review by jmbowman — APPROVED** [review]
If you find yourself needing to do more things like this, there's a codemod utility at https://github.com/adamchainz/django-upgrade that can do most of it for you (Arbi-BOM is currently running it across edx-platform).

File: auth_backends/urls.py (hunk 1 above)
Comment (christopappas) [reply auth_backends/urls.py]: sounds good, will change

**Pull request merged**